src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs

```diff
@@ -0,0 +1,153 @@
+
+                SystemEvents.KillTimer(timer);
+                elapsed.Reset();
+
+                // Timer should not fire once killed
+                Assert.False(elapsed.WaitOne(TimerInterval * SystemEventsTest.UnexpectedEventMultiplier));
+            }
+            finally
+            {
+                SystemEvents.TimerElapsed -= handler;
+            }
+        }
+
+        [Fact]
+        public void ConcurrentTimers()
+        {
+            const int numConcurrentTimers = 10;
+            var timersSignalled = new Dictionary<IntPtr, bool>();
+
+            TimerElapsedEventHandler handler = (sender, args) =>
+            {
+                if (timersSignalled.ContainsKey(args.TimerId))
+                {
+                    timersSignalled[args.TimerId] = true;
+                }
+            };
+
+            SystemEvents.TimerElapsed += handler;
+            try
+            {
+                for (int i = 0; i < numConcurrentTimers; i++)
+                {
+                    timersSignalled[SystemEvents.CreateTimer(TimerInterval)] = false;
+                }
+
+                Thread.Sleep(TimerInterval * SystemEventsTest.ExpectedEventMultiplier);
+
+                foreach(var timer in timersSignalled.Keys.ToArray())
+                {
+                    Assert.True(timersSignalled[timer]);
+                    SystemEvents.KillTimer(timer);
+                }
+            }
+            finally
+            {
+                SystemEvents.TimerElapsed -= handler;
+            }
+        }
+
+        [OuterLoop]
+        [Theory]
+        [InlineData(500)]  // .5s
+        [InlineData(1000)]  // 1s
+        [InlineData(2000)]  // 2s
+        [InlineData(10000)] // 10s
+        [InlineData(30000)] // 30s
+        public void TimerElapsedIsRoughlyEquivalentToInterval(int interval)
+        {
+            const double permittedPercentDifference = 0.1;
+            var elapsed = new AutoResetEvent(false);
+            IntPtr timer = IntPtr.Zero;
+            var stopwatch = new Stopwatch();
+
+            TimerElapsedEventHandler handler = (sender, args) =>
+            {
+                if (args?.TimerId == timer)
+                {
+                    stopwatch.Stop();
+                    elapsed.Set();
+                }
+            };
+
+            SystemEvents.TimerElapsed += handler;
+            try
+            {
+                timer = SystemEvents.CreateTimer(interval);
+                stopwatch.Start();
+                Assert.True(elapsed.WaitOne(interval * SystemEventsTest.ExpectedEventMultiplier));
+
+                var percentDifference = (double)Math.Abs(stopwatch.ElapsedMilliseconds - interval) / interval;
+                Assert.True(percentDifference < permittedPercentDifference, $"Timer should fire within {permittedPercentDifference * 100.0}% of expected interval {interval}, actual: {stopwatch.ElapsedMilliseconds}, difference: {percentDifference * 100.0}%");
```

Comment:
This test is likely to fail periodically in official runs due to concurrent tests running and other activity. It should not be _less_ than the interval (although I think I have seen cases where tests seemingly slept less than requested) so you could check for 10% under 50% over or something.

Reply:
I think there are some tests that have a 1 sec delay that I ended up cranking up to 30 secs.. to give you an idea. One notch at atime.